### PR TITLE
Make TotalRequests in workload.Info a list instead of map

### LINKED ISSUE #18: Preserve order of podsets in workload info
[TotalRequests](https://github.com/kubernetes-sigs/kueue/blob/b86eeb134e3721de9428eac99e286fe552ae72ee/pkg/workload/workload.go#L32) in workload.Info is currently a map, when iterating over it to assign resources, we will loose the original order of the podsets. 

When scheduling, a podset gets assigned flavors depending on the iteration order of this map, and so the assignment will not be deterministic. 

This actually caused a flake in the following test: https://github.com/kubernetes-sigs/kueue/blob/b86eeb134e3721de9428eac99e286fe552ae72ee/pkg/scheduler/scheduler_test.go#L406

/kind bug

### PR DESCRIPTION
**Pull request by ahg-g**
#### What type of PR is this?

/kind bug

#### Which issue(s) this PR fixes:
Fixes #18

### CODE REVIEW

File: pkg/workload/workload.go [hunk 1]
```diff
@@ -29,12 +29,13 @@ import (
 type Info struct {
 	Obj *kueue.QueuedWorkload
 	// maps PodSet name to total resources requested by the set.
```
Comment (denkensk) [inline pkg/workload/workload.go:31]: update the comment?

**Comment by alculquicondor** [comment]
lgtm after Alex's comment is addressed.

File: pkg/workload/workload.go (hunk 1 above)
Comment (ahg-g) [reply pkg/workload/workload.go]: done.

**Pull request merged**